### PR TITLE
Allow set filter query param directly

### LINKED ISSUE #30: Add ability to set more complex filter query parameter
Accordingly JsonApi specification 

> JSON:API is agnostic about the strategies supported by a server. The filter query parameter can be used as the basis for any number of filtering strategies.

Current JsonApiRequestBuilder realization restrict filter strategy https://github.com/woohoolabs/yang/blob/master/src/JsonApi/Request/JsonApiRequestBuilder.php#L258

The filter params can be represented only like `filter[field1]=value&filter[field2]=val1,val2`  

But my strategy requires a more flexible form like `filter[field1][like]=foo&filter[field2][lt]=100&filter[field2][gt]=5`  (or even with more condition depth)

Right now is not possible even to override this method, because queryString property is private

If you are interested in solving this problem, we can discuss ways to fix it and I can create a PR

### PR DESCRIPTION
**Pull request by Insolita**
Fix #30 

- add composer:v1 tag for GitHub workflow (composer v2 requires update dependencies)
- fix broken test https://github.com/woohoolabs/yang/blob/master/tests/JsonApi/Request/JsonApiRequestBuilderTest.php#L103, provide really invalid uri
- csfix

### CODE REVIEW

**Comment by kocsismate** [comment]
Thank you for the PR! I'll try to look at it in the upcoming days!

File: src/JsonApi/Request/JsonApiRequestBuilder.php [hunk 1]
```diff
@@ -371,7 +350,8 @@ private function getMediaTypeWithProfiles(array $profiles, bool $compatibility):
         $result = "application/vnd.api+json";
 
         if (empty($profiles) === false) {
-            $result .= ';profile="' . implode(" ", $profiles) . '"' . ($compatibility ? ',application/vnd.api+json' : "");
+            $result .= ';profile="' . implode(" ", $profiles) . '"' . ($compatibility ? ',application/vnd.api+json'
+                    : "");
```
Comment (kocsismate) [inline src/JsonApi/Request/JsonApiRequestBuilder.php:354]: Is this an intentional change?

**Review by kocsismate — COMMENTED** [review]
Looks good to me except for the single tiny comment below.

File: src/JsonApi/Request/JsonApiRequestBuilder.php (hunk 1 above)
Comment (Insolita) [reply src/JsonApi/Request/JsonApiRequestBuilder.php:354]: It has been changed after apply csfixer

Comment (Insolita) [reply src/JsonApi/Request/JsonApiRequestBuilder.php]: But looks a little bit weird. I've fix it

**Comment by kocsismate** [comment]
I've just merged your changes! :)

**Comment by Insolita** [comment]
What about adding a new tag?

**Comment by kocsismate** [comment]
Done!

**Pull request merged**